tests/vmi_networking_test.go

```diff
@@ -93,279 +92,264 @@ var _ = Describe("Networking", func() {
 		expecter, err := expecterFactory(vmi)
 		Expect(err).ToNot(HaveOccurred())
 		expecter.Close()
+		return vmi
+	}
+
+	startTCPServer := func(vmi *v1.VirtualMachineInstance, port int) {
+		expecter, err := tests.LoggedInCirrosExpecter(vmi)
+		Expect(err).ToNot(HaveOccurred())
+		defer expecter.Close()
+
+		resp, err := expecter.ExpectBatch([]expect.Batcher{
+			&expect.BSnd{S: "\n"},
+			&expect.BExp{R: "\\$ "},
+			&expect.BSnd{S: fmt.Sprintf("screen -d -m nc -klp %d -e echo -e \"Hello World!\"\n", port)},
+			&expect.BExp{R: "\\$ "},
+			&expect.BSnd{S: "echo $?\n"},
+			&expect.BExp{R: "0"},
+		}, 60*time.Second)
+		log.DefaultLogger().Infof("%v", resp)
+		Expect(err).ToNot(HaveOccurred())
 	}
 
 	// TODO this is not optimal, since the one test which will initiate this, will look slow
 	tests.BeforeAll(func() {
 		tests.BeforeTestCleanup()
 
-		prepareVMIs := func(withPodNetwork bool) (*v1.VirtualMachineInstance, *v1.VirtualMachineInstance) {
```

Comment:
We have several configurations that would each benefit from validating connectivity with the whole suite. You kill the boolean flag here. How should we approach it now?

Reply:
So the only thing which it extra tested, if the pod network was explicitly requested on vm-creation, compared to implicit creation. For this, I added an extra test, instead of running the whole suite again.

Reply:
Yeah but we may have more variations to test in the future (actually, I am adding one for custom MAC addresses in another PR, so my question is not theoretical).

Reply:
I see. In this case I would assume one extra test which starts a vm with the mac set. Then log into the vm to see if  it got the mac and finally try to access it via a pod based on the reported ip for the vm. I think in most cases we can cover that with one explicit extra test to ensure proper connectivity in general for the new feature.